pkg/kubectl/cmd/expose.go

```diff
@@ -149,9 +149,9 @@ func RunExpose(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 	// For objects that need a pod selector, derive it from the exposed object in case a user
 	// didn't explicitly specify one via --selector
 	if s, found := params["selector"]; found && kubectl.IsZero(s) {
-		s, err := f.PodSelectorForObject(inputObject)
+		s, err := f.MapBasedSelectorForObject(inputObject)
```

Comment:
@kargakis Since this is in pkg/kubectl/... and an exported function, we weren't sure who else outside Kubernetes was using this, particularly OpenStack. So we left the existing function as is. If you are certain that OpenStack doesn't use the factory, we can probably eliminate one of the two functions.

cc @bgrant0607